features/net/network-socket/Socket.h

```diff
@@ -182,6 +182,8 @@ class Socket {
         attach(mbed::callback(obj, method));
     }
 
+    NetworkStack *stack() const { return _stack; }
```

Comment:
The reason it was added is no longer valid, so it no longer needs to be added for this commit. (It was needed to implement the DNS work in the first draft.)

But I remain unconvinced of the merit of hiding NetworkStack.